library/src/main/kotlin/com/connectrpc/impl/ProtocolClient.kt

```diff
@@ -129,8 +144,16 @@ class ProtocolClient(
                 )
             }
             return cancelable
-        } catch (e: Exception) {
-            throw e
```

Comment:
This was the other uncaught exception. 🤦
Since this is an async operation, instead of throwing, it needs to invoke the callback with an error.